uabot.code-snippets

```diff
@@ -0,0 +1,741 @@
+			"\t\"type\": \"Search\",\r",
+			"\t\"arguments\": {\r",
+			"\t\t\"queryText\": \"${1:The query to send}\"\r",
+			"\t}\r",
+			"},$0"
+		],
+		"description": "Coveo UABot Custom Query Search Event"
+	},
+
+	"UABot Complete Click Event": {
+		"prefix": "uaboteventclick",
+		"scope": "plaintext,json",
+		"body": [
+			"{\r",
+			"\t\"type\": \"Click\",\r",
+			"\t\"arguments\": {\r",
+			"\t\t\"docNo\": ${1:The rank of the document to click (base: 0, random: -1)},\r",
+			"\t\t\"probability\": ${2:The probability that the user will click (between 0 and 1)},\r",
+			"\t\t\"offset\": ${3:Optional. Offset used in random document clicking (integer)},\r",
+			"\t\t\"quickview\": ${4:Optional. If the click is a quickview or not. true/false (default, false)},\r",
+			"\t\t\"fakeClick\": ${5:Optional. true/false},\r",
+			"\t\t\"fakeResponse\": ${6:Optional},\r",
+			"\t\t\"customData\": {\r",
+			"\t\t\t$0\r",
+			"\t\t}\r",
+			"\t}\r",
+			"},"
+		],
+		"description": "Coveo UABot Complete Click Event"
+	},
+
+	"UABot Precise Click Event": {
+		"prefix": "uaboteventclick",
+		"scope": "plaintext,json",
+		"body": [
+			"{\r",
+			"\t\"type\": \"Click\",\r",
+			"\t\"arguments\": {\r",
+			"\t\t\"docNo\": ${1:The rank of the document to click (base: 0)},\r",
+			"\t\t\"probability\": ${2:The probability that the user will click (between 0 and 1)},\r",
+			"\t\t\"quickview\": ${4:Optional. If the click is a quickview or not. true/false (default, false)}\r",
+			"\t}\r",
+			"},$0"
+		],
+		"description": "Coveo UABot Precise Click Event"
+	},
+
+	"UABot Random Click Event": {
+		"prefix": "uaboteventclick",
+		"scope": "plaintext,json",
+		"body": [
+			"{\r",
+			"\t\"type\": \"Click\",\r",
+			"\t\"arguments\": {\r",
+			"\t\t\"docNo\": -1,\r",
+			"\t\t\"probability\": ${2:The probability that the user will click (between 0 and 1)},\r",
+			"\t\t\"offset\": ${3:Offset used in random document clicking (integer)},\r",
+			"\t\t\"quickview\": ${4:Optional. If the click is a quickview or not. true/false (default, false)}\r",
+			"\t}\r",
+			"},$0"
+		],
+		"description": "Coveo UABot Random Click Event"
+	},
+
+	"UABot Complete SearchAndClick Event": {
+		"prefix": "uaboteventsearchandclick",
+		"scope": "plaintext,json",
+		"body": [
+			"{\r",
+			"\t\"type\": \"SearchAndClick\",\r",
+			"\t\"arguments\": {\r",
+			"\t\t\"queryText\": \"${1:The query to send}\",\r",
+			"\t\t\"caseSearch\": ${2:Optional. If the query comes from a Case Creation interface. true/false},\r",
+			"\t\t\"inputTitle\": \"${3:Optional. The title of the input that triggered the search if it was a case search}\",\r",
+			"\t\t\"docClickTitle\": \"${4:Optional. The title of the document you want to click on.}\",\r",
+			"\t\t\"matchField\": \"${5:Optional. When omitting docClickTitle. The name of the field for the match expression (Without @).}\",\r",
+			"\t\t\"matchPattern\": \"${6:Optional. When omitting docClickTitle. The value for the match expression.}\",\r",
```

Comment:
I would make two distinct snippets, one with matchField/matchPattern and one with a specific title.